src/test/java/com/meilisearch/integration/SettingsTest.java

```diff
@@ -0,0 +1,183 @@
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		SettingsRequest settings = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+
+		settings.setRankingRules(new String[]{
+			"typo",
+			"words",
+			"proximity",
+			"attribute",
+			"wordsPosition",
+			"exactness",
+			"desc(release_date)",
+			"desc(rank)"});
+
+		index.updateSettings(settings);
+		Thread.sleep(1000);
+
+		SettingsRequest newSettings = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+
+		assertEquals(8, newSettings.getRankingRules().length);
+	}
+
+	/**
+	 * Test update settings changing the synonyms
+	 */
+	@Test
+	public void testUpdateSettingsSynonyms() throws Exception {
+		String indexUid = "updateSettingsSynonyms";
+		Index index = client.index(indexUid);
+		GsonJsonHandler jsonGson = new GsonJsonHandler();
+
+		TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+		UpdateStatus updateInfo = jsonGson.decode(
+			index.addDocuments(testData.getRaw()),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		SettingsRequest settings = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+
+		HashMap<String, String[]> synonyms = new HashMap();
+		synonyms.put("wolverine", new String[] {"xmen", "logan"});
+		synonyms.put("logan", new String[] {"wolverine"});
+		settings.setSynonyms(synonyms);
+
+		index.updateSettings(settings);
+		Thread.sleep(1000);
+
+		SettingsRequest newSettings = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+
+		assertEquals(2, newSettings.getSynonyms().size());
+	}
+
+	/**
+	 * Test update settings changing the synonyms
+	 */
+	@Test
+	public void testResetSettings() throws Exception {
+		String indexUid = "updateSettingsSynonyms";
+		Index index = client.index(indexUid);
+		GsonJsonHandler jsonGson = new GsonJsonHandler();
+
+		TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+		UpdateStatus updateInfo = jsonGson.decode(
+			index.addDocuments(testData.getRaw()),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		SettingsRequest settings = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+
+		HashMap<String, String[]> synonyms = new HashMap();
+		synonyms.put("wolverine", new String[] {"xmen", "logan"});
+		synonyms.put("logan", new String[] {"wolverine"});
+		settings.setSynonyms(synonyms);
+
+		index.updateSettings(settings);
+		Thread.sleep(1000);
+		SettingsRequest newSettings = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+
+		assertEquals(2, newSettings.getSynonyms().size());
+
+		index.resetSettings();
+		Thread.sleep(1000);
+		SettingsRequest settingsAfterReset = jsonGson.decode(
+			index.getSettings(),
+			SettingsRequest.class
+		);
+		assertEquals(0, settingsAfterReset.getSynonyms().size());
```

Comment:
For now, the default is 0, but I don't know if this could ever change. What if, instead of comparing it to 0, you compare that the size is the same it returned before modifying the settings? (Meaning, comparing it to the length you got on L155)